### ISSUE TITLE
Stop sending heartbeat to peers on unreachable store

### ISSUE THREAD

**Issue by Connor1996** [tikv/tikv#13724]
## Development Task

After https://github.com/tikv/tikv/pull/13677, we use `wait_for_connected` to actively detect the connection to unreachable stores. With that, the leader don't need to send heartbeat to peer on unreachable store periodically to detect whether the connection is connected again. Once reconnected, we can simply broadcast a message to inform them. It can save a lot of heartbeat messages and unreachable messages for individual regions triggered by `clear_pending_message`.

**Comment by BusyJay** [comment]
Good idea! But instead of heartbeat messages, I think the key point here is to allow leader to hibernate until the store is reconnected again. This should reduce a lot of CPU costs.

**Comment by Connor1996** [comment]
It can't hibernate. It may have writes and need to replicate to other peers.

**Comment by BusyJay** [comment]
> It can't hibernate. It may have writes and need to replicate to other peers.

Of course wait till reachable peers catches up all states. See #13396.
